homeassistant/components/ecobee/number.py

```diff
@@ -0,0 +1,108 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class EcobeeNumberEntityDescriptionBase:
+    """Required values when describing Ecobee number entities."""
+
+    ecobee_setting_key: str
+    set_fn: Callable[[EcobeeData, int, int], Awaitable]
+
+
+@dataclass
+class EcobeeNumberEntityDescription(
+    NumberEntityDescription, EcobeeNumberEntityDescriptionBase
+):
+    """Class describing Ecobee number entities."""
+
+
+VENTILATOR_NUMBERS = (
+    EcobeeNumberEntityDescription(
+        key="home",
+        name="home",
+        ecobee_setting_key="ventilatorMinOnTimeHome",
+        set_fn=lambda data, id, min_time: data.ecobee.set_ventilator_min_on_time_home(
+            id, min_time
+        ),
+    ),
+    EcobeeNumberEntityDescription(
+        key="away",
+        name="away",
+        ecobee_setting_key="ventilatorMinOnTimeAway",
+        set_fn=lambda data, id, min_time: data.ecobee.set_ventilator_min_on_time_away(
+            id, min_time
+        ),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the ecobee thermostat number entity."""
+    data = hass.data[DOMAIN]
+    entities = []
+    _LOGGER.debug("Adding min time ventilators numbers (if present)")
+    for index, thermostat in enumerate(data.ecobee.thermostats):
+        if thermostat["settings"]["ventilatorType"] == "none":
+            continue
+        _LOGGER.debug("Adding %s's ventilator min times number", thermostat["name"])
+        for numbers in VENTILATOR_NUMBERS:
+            entities.append(EcobeeVentilatorMinTime(data, index, numbers))
+
+    async_add_entities(entities, True)
+
+
+class EcobeeVentilatorMinTime(EcobeeBaseEntity, NumberEntity):
+    """A number class, representing min time  for an ecobee thermostat with ventilator attached."""
+
+    _attr_native_min_value = 0
+    _attr_native_max_value = 60
+    _attr_native_step = 5
+    _attr_native_unit_of_measurement = UnitOfTime.MINUTES
+    _attr_has_entity_name = True
+
+    def __init__(self, data, thermostat_index, description):
+        """Initialize ecobee ventilator platform."""
+        super().__init__(data.ecobee.get_thermostat(thermostat_index))
+
+        self.data = data
+        self.thermostat_index = thermostat_index
+        self.ecobee_setting_key = description.ecobee_setting_key
+        self.set_fn = description.set_fn
+        self._attr_name = f"Ventilator min time {description.name}"
+        self._attr_unique_id = (
+            f'{self.thermostat["identifier"]}_ventilator_{description.key}'
+        )
+        self._attr_native_value = self.thermostat["settings"][
+            description.ecobee_setting_key
+        ]
+
+    async def async_update(self) -> None:
+        """Get the latest state from the thermostat."""
+        await self.data.update()
+        self.thermostat = self.data.ecobee.get_thermostat(self.thermostat_index)
```

Comment:
I was thinking about how to avoid this.  

I think we could make thermostat a property that wrapped this code and than we wouldn't need to set it in init

Reply:
Would making it a property make it available to consume by the user? (I don't think we want that)

Reply:
No, the user doesn't have direct access to the entity code

Reply:
Something like that ?? 
```python 
@property
    def thermostat_data(self):
        """get the latest cache data from the thermostat"""
        return self.data.ecobee.get_thermostat(self.thermostat_index)

    async def async_update(self) -> None:
        """Get the latest state from the thermostat."""
        await self.data.update()
        self._attr_native_value = self.thermostat_data["settings"][
            self.ecobee_setting_key
        ]   
```

Reply:
I was thinking something like this. It should cherry-pick cleanly, but since I don't have an ecobee that ventilates I can't test it.

https://github.com/bdraco/home-assistant/commit/74f1bcfdc80d4dcab5f2547b81ec32035a223ab4

Reply:
Ok I cherry picked, and tested. It works perfectly. For some reason Github is not adding it to the PR, but it is in my fork,. https://github.com/marcolivierarsenault/core/commits/add_ecobee_ventilator 

I guess, GH will sync later